engines/bladerunner/ui/ui_input_box.cpp

```diff
@@ -88,9 +88,10 @@ void UIInputBox::hide() {
 }
 
 void UIInputBox::handleKeyDown(const Common::KeyState &kbd) {
+	char kc = Common::U32String(Common::String::format("%c", kbd.ascii), Common::kISO8859_1).encode(Common::kDos850).firstChar();
```

Comment:
can you add some description & possible examples to a comment as this looks very confusing

Reply:
Done. It is kind of difficult to test, since UIInputBox::charIsValid() will block all these characters anyway (regardless of encoding). If you want to test it you have to comment out the checks in there, in particular the `kc <= '~'`.
After that it will basically depend on your locale and your keyboard which extra characters you can test (I don't know where you're located). I have tested the 'äüöß' characters. The font does have glyphs for it, even in the English version of the game.

Reply:
Looks okay, but SDL is only one backend (and I didn't know that SDL do conversion to 8859-1), there are more like Android or IOS.
Do we really need this feature - support for national character entry in-game? Its adding unnecessary complexity a possible issues for other backends.
Maybe we can omit this one, in ScummVM's GUI is okay, but for game... I don't think benefits surpasses the possible issues and more complex code.